dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBMongoDB/AzureCosmosDBMongoDBMemoryStore.cs

```diff
@@ -0,0 +1,571 @@
+    private readonly String _kind;
+    private readonly int _numLists;
+    private readonly String _similarity;
+    private readonly int _dimensions;
+    private readonly int _numberOfConnections;
+    private readonly int _efConstruction;
+    private readonly int _efSearch;
+
+    /// <summary>
+    /// Initiates a AzureCosmosDBMongoDBMemoryStore instance using a Azure CosmosDB Mongo vCore connection string
+    /// and other properties required for vector search.
+    /// </summary>
+    /// <param name="connectionString">Connection string required to connect to Azure Cosmos Mongo vCore.</param>
+    /// <param name="databaseName">Database name for Mongo vCore DB</param>
+    /// <param name="indexName">Index name for the Mongo vCore DB</param>
+    /// <param name="applicationName">Application name for the client for tracking and logging</param>
+    /// <param name="kind">Kind: Type of vector index to create.
+    ///     Possible options are:
+    ///         - vector-ivf
+    ///         - vector-hnsw: available as a preview feature only,
+    ///                        to enable visit https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/preview-features</param>
+    /// <param name="numLists">This integer is the number of clusters that the inverted file (IVF) index uses to group the vector data.
+    /// We recommend that numLists is set to documentCount/1000 for up to 1 million documents and to sqrt(documentCount)
+    /// for more than 1 million documents. Using a numLists value of 1 is akin to performing brute-force search, which has
+    /// limited performance.</param>
+    /// <param name="similarity">Similarity metric to use with the IVF index.
+    ///     Possible options are:
+    ///         - COS (cosine distance),
+    ///         - L2 (Euclidean distance), and
+    ///         - IP (inner product).</param>
+    /// <param name="dimensions">Number of dimensions for vector similarity. The maximum number of supported dimensions is 2000.</param>
+    /// <param name="numberOfConnections">The max number of connections per layer (16 by default, minimum value is 2, maximum value is
+    /// 100). Higher m is suitable for datasets with high dimensionality and/or high accuracy requirements.</param>
+    /// <param name="efConstruction">The size of the dynamic candidate list for constructing the graph (64 by default, minimum value is 4,
+    /// maximum value is 1000). Higher ef_construction will result in better index quality and higher accuracy, but it will
+    /// also increase the time required to build the index. EfConstruction has to be at least 2 * m.</param>
+    /// <param name="efSearch">The size of the dynamic candidate list for search (40 by default). A higher value provides better recall at
+    /// the cost of speed.</param>
+    public AzureCosmosDBMongoDBMemoryStore(
+        string connectionString,
+        string databaseName,
+        string? indexName = "default_index",
```

Comment:
Consider adding a separate Options class for all the optional options, so that it's easy to add more or deprecate obsolete ones without needing to update this signature all the time.  See e.g. how CosmosClient does it:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.cosmosclient.-ctor?view=azure-dotnet#microsoft-azure-cosmos-cosmosclient-ctor(system-string-microsoft-azure-cosmos-cosmosclientoptions)

Reply:
Added a AzureCosmosDBMongoDBConfig which has all the vector search properties.